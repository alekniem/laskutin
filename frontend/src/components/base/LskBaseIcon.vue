<template>
    <v-tooltip bottom>
        <template #activator="{ attrs, on }">
            <v-icon
                @click="clickFunction(parameter)"
                :class="cssClass"
                small
                v-bind="attrs"
                v-on="on"
            >
                {{ iconName }}
            </v-icon>
        </template>
        <span>{{ title | trans }}</span>
    </v-tooltip>
</template>

<script>
export default {
    name: "LskBaseIcon",

    props: {
        clickFunction: { type: Function, required: true },
        cssClass: { type: String, required: false },
        iconName: { type: String, required: true },
        parameter: { type: Object, required: true },
        title: { type: String, required: true },
    },
};
</script>
