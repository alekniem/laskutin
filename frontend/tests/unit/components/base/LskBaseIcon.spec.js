import flushPromises from "flush-promises";
import sinon from "sinon";
import LskBaseIcon from "@/components/base/LskBaseIcon";
import { createWrapper } from "UT/lsk_test_helper";

describe("LskBaseIcon", () => {
    afterEach(() => {
        sinon.restore();
    });

    it("should match snapshot", () => {
        const wrapper = createWrapper(LskBaseIcon, {
            propsData: {
                clickFunction: (param) => param,
                cssClass: "mr-3",
                iconName: "mdi-pencil",
                parameter: {},
                title: "TEST TITLE",
            },
        });
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("should call function when icon is clicked", async () => {
        const spy = sinon.spy();
        const param = { test: "TEST" };
        const wrapper = createWrapper(LskBaseIcon, {
            propsData: {
                clickFunction: spy,
                iconName: "mdi-pencil",
                parameter: param,
                title: "TEST TITLE",
            },
        });
        await flushPromises();
        wrapper.find("button").trigger("click");
        await flushPromises();
        expect(spy.calledWith(param)).toBe(true);
    });
});
