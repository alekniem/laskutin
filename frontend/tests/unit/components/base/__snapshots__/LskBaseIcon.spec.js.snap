// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LskBaseIcon should match snapshot 1`] = `
<span class="v-tooltip v-tooltip--bottom">
  <button type="button"
          aria-haspopup="true"
          aria-expanded="false"
          class="v-icon notranslate v-icon--link mdi mdi-pencil theme--light mr-3"
          style="font-size: 16px;"
  >
  </button>
</span>
`;
